lib/oli_web/live/delivery/instructor_dashboard/instructor_dashboard_live.ex

```diff
@@ -317,4 +322,45 @@ defmodule OliWeb.Delivery.InstructorDashboard.InstructorDashboardLive do
       })
     end)
   end
+
+  defp add_enrollment_and_payment_data(students, section) do
+    Enum.map(students, fn student ->
+      Map.merge(
+        student,
+        %{
+          enrollment_status:
+            if(Sections.is_enrolled?(student.id, section.slug), do: "Enrolled", else: "Suspended"),
```

Comment:
Sections.is_enrolled, Paywall.summarize_access, Section.get_enrollment all make database calls, numerous ones.  This cannot be executed within the map over all students.  We need to instead DB efficient Bulk versions of methods that will provide a constant number of queries over the enrolled students.

Reply:
Sure, you're right. I'll fix it.